imageregistry/v1/types.go

```diff
@@ -306,6 +306,78 @@ type ImageRegistryConfigStorageIBMCOS struct {
 	ServiceInstanceCRN string `json:"serviceInstanceCRN,omitempty"`
 }
 
+// EndpointAccessibility defines the Alibaba VPC endpoint for storage
+type EndpointAccessibility string
+
+// AlibabaEncryptionMode defines an enumerable type for the encryption mode
+type AlibabaEncryptionMode string
+
+const (
+	// InternalEndpoint sets the VPC endpoint to internal
+	InternalEndpoint EndpointAccessibility = "Internal"
+	// PublicEndpoint sets the VPC endpoint to public
+	PublicEndpoint EndpointAccessibility = "Public"
+
+	// ClearText is an AlibabaEncryptionMode. This means no encryption
+	ClearText AlibabaEncryptionMode = "ClearText"
+	// AES256 is an AlibabaEncryptionMode. This means AES256 encryption
+	AES256 AlibabaEncryptionMode = "AES256"
+	// KMS is an AlibabaEncryptionMode. This means KMS encryption
+	KMS AlibabaEncryptionMode = "KMS"
+)
+
+// EncryptionAlibaba this a union type in kube parlance.  Depending on the value for the encryptionType,
+// different pointers may be used
+type EncryptionAlibaba struct {
+	// EncryptionType defines the different encrytion types available
+	// Empty value means no opinion and the platform chooses the a default, which is subject to change over time.
+	// Currently the default is `ClearText`.
+	// +kubebuilder:validation:Enum="ClearText"; "KMS"; "AES256"
+	// +optional
+	Type AlibabaEncryptionMode `json:"encryptionType"`
+
+	// KMS (key management service) is an encryption type provided to encrypt the provider
+	// +optional
+	KMS *KMSEncryptionAlibaba `json:"kms,omitempty"`
+}
+
+type KMSEncryptionAlibaba struct {
+	// KeyID holds the KMS encryption key ID
+	// +kubebuilder:validation:Required
+	KeyID string `json:"keyID"`
```

Comment:
can this be empty? Which format?

Reply:
When the `Type` is set to `KMS` then a key would be required.  When it is not, it can be empty.
```
The CMK ID that must be specified when SSEAlgorithm is set to KMS and a specified CMK is used for encryption. In other cases, this element must be set to null.
```